codegen/src/test/resources/q4/Job.scala

```diff
@@ -0,0 +1,84 @@
+import org.apache.flink.api.java.utils.ParameterTool
+import org.apache.flink.core.fs.FileSystem
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.streaming.api.scala._
+import org.hkust.RelationType.Payload
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.apache.flink.util.Collector
+object Job {
+   val lineitemTag: OutputTag[Payload] = OutputTag[Payload]("lineitem")
+   val ordersTag: OutputTag[Payload] = OutputTag[Payload]("orders")
+   def main(args: Array[String]) {
+      val env = StreamExecutionEnvironment.getExecutionEnvironment
+      val params: ParameterTool = ParameterTool.fromArgs(args)
+      env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+      var executionConfig = env.getConfig
+      executionConfig.enableObjectReuse()
+      val inputpath = "file:///aju/q3flinkInput.csv"
+      val outputpath = "file:///aju/q3flinkOutput.csv"
+      val inputStream : DataStream[Payload] = getStream(env,inputpath)
+      val orders : DataStream[Payload] = inputStream.getSideOutput(ordersTag)
+      val lineitem : DataStream[Payload] = inputStream.getSideOutput(lineitemTag)
+      val ordersS = orders.keyBy(i => i._3)
+      .process(new Q4OrdersProcessFunction())
+      val lineitemS = ordersS.connect(lineitem)
+      .keyBy(i => i._3, i => i._3)
+      .process(new Q4LineitemProcessFunction())
+      val result = lineitemS.keyBy(i => i._3)
+      .map(x => Payload("Aggregate", "Addition", x._3, x._4, x._5, x._6))
+      .keyBy(i => i._3)
+      .process(new Q4Aggregate2ProcessFunction())
+      .writeAsText(outputpath,FileSystem.WriteMode.OVERWRITE)
```

Comment:
There should be a .map(x => (x._4.mkString(", "), x._5.mkString(", "), x._6)) after line 30.